src/openforms/registrations/contrib/objects_api/config.py

```diff
@@ -453,20 +458,40 @@ def _validate_catalogue_and_document_types(attrs: RegistrationOptions) -> None:
         "informatieobjecttype_attachment",
     ):
         url = attrs.get(field)
-        if url:
-            warnings.warn(
-                "URL references to document types are deprecated and will be remove in "
-                "Open Forms 3.0",
-                DeprecationWarning,
-            )
-        if url and url not in informatieobjecttypen_urls:
-            err_tpl = (
-                _("The provided {field} does not exist in the Catalogi API.")
-                if catalogus is None
-                else _("The provided {field} does not exist in the selected catalogue.")
-            )
-            err_msg = err_tpl.format(field=field)
-            raise serializers.ValidationError({field: err_msg}, code="not-found")
+        if not url:
+            continue
+
+        warnings.warn(
+            "URL references to document types are deprecated and will be remove in "
+            "Open Forms 3.0",
+            DeprecationWarning,
+        )
+
+        err_tpl = (
+            _("The provided {field} does not exist in the Catalogi API.")
+            if catalogus is None
+            else _("The provided {field} does not exist in the selected catalogue.")
+        )
+        err_msg = err_tpl.format(field=field)
+
+        if informatieobjecttypen_urls:
+            if url not in informatieobjecttypen_urls:
+                raise serializers.ValidationError({field: err_msg}, code="not-found")
+        else:
+            assert api_group.catalogi_service
+            iotypen_endpoint = (
+                furl(api_group.catalogi_service.api_root) / "informatieobjecttypen"
+            ).url
+
+            if not url.startswith(iotypen_endpoint):
+                raise serializers.ValidationError({field: err_msg}, code="not-found")
+            with get_catalogi_client(api_group) as client:
+                response = client.head(url)
```

Comment:
I think we can do this only if we know that `client.api_version >= (1, 1, 0)` since in 1.0 the `HEAD` method is not required to be provided.

```suggestion
                response = client.head(url) if client.api_version >= (1, 1, 0) else client.get(url)
```